os/arch/arm/src/armv8-m/mpu.h

```diff
@@ -687,6 +687,29 @@ static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb)
 #define up_set_mpu_app_configuration(x)
 #endif
 
+#if defined(CONFIG_MPU_STACKGUARD)
+static inline void up_set_mpu_region_configuration(struct tcb_s *rtcb, int region_no)
+{
+	/* We update the MPU registers only if:
+	 * This is not a kernel thread AND
+	 * It has a non zero value of base address (This ensures valid MPU setting)
+	 */
+	int i = 3 * region_no;
+	if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_KERNEL) {
+		return;
+	}
+
+	if (rtcb->mpu_regs[i + REG_RBAR]) {
+		putreg32(rtcb->mpu_regs[i + REG_RNR], MPU_RNR);
+		putreg32(rtcb->mpu_regs[i + REG_RBAR], MPU_RBAR);
+		putreg32(rtcb->mpu_regs[i + REG_RASR], MPU_RASR);
+	}
+
+}
+#else
+#define up_set_mpu_region_configuration(x, y)
```

Comment:
When you want to call this function without the conditional, this is right. But you call this with conditional at every locations, then this is unnecessary.

Reply:
done